core/src/commonMain/kotlin/entity/interaction/Interaction.kt

```diff
@@ -24,10 +25,13 @@ public sealed interface Interaction : InteractionBehavior {
 
     override val applicationId: Snowflake get() = data.applicationId
 
-    override val channelId: Snowflake get() = data.channelId
+    override val channelId: Snowflake get() = data.channelId.value!!
 
     override val token: String get() = data.token
 
+    /** A Partial [Channel] object for the interaction. */
+    public val partialChannel: Channel? get() = data.channel.value?.let { Channel(it, kord, supplier) }
```

Comment:
should this maybe use `Channel.from()`? or should we only expose the channel type as a property (as it's the only field guaranteed to be present)?

Reply:
the behavior of the channel is already accessible through `InteractionBehavior.channel`, so that would also give a cleaner api